data_validation/__main__.py

```diff
@@ -472,10 +501,284 @@ def store_yaml_config_file(args, config_managers):
     cli_tools.store_validation(config_file_path, yaml_configs)
 
 
-def run(args):
-    """ """
-    config_managers = build_config_managers_from_args(args)
+def partition_and_store_config_files(args: Namespace) -> None:
+    """Build multiple YAML Config files using user specified partition logic
+
+    Args:
+        args (Namespace): User specified Arguments
+        config_managers (List[ConfigManager]): List of config manager instances.
+
+    Returns:
+        None
+    """
+    if args.validate_cmd == "row":
+        config_managers = build_config_managers_from_args(args)
+        partition_configs(args, config_managers)
+    else:
+        raise ValueError(f"Validation Argument '{args.validate_cmd}' is not supported")
+
+
+def partition_configs(args: Namespace, config_managers: List[ConfigManager]):
+    """Takes a list of ConfigManager object and splits each it into multiple
+    ConfigManager objects applying supplied partition logic.
+
+    Args:
+        args (Namespace): User specified Arguments.
+        config_managers (List[ConfigManager]): List of config manager instances.
+
+    Returns:
+        None
+    """
+
+    config_dir = _get_arg_config_dir(args)
+    partition_type = _get_arg_partition_type(args)
+    if partition_type == "primary_key":
+        partition_filters = _get_primary_key_partition_filters(args)
+    elif partition_type == "primary_key_mod":
+        # TODO: Add support for Primary_key + Mod
+        raise ValueError(f"Partition Type: '{partition_type}' is not supported")
+    elif partition_type == "hash_mod":
+        # TODO: Add support for Hash + Mod
+        raise ValueError(f"Partition Type: '{partition_type}' is not supported")
+
+    _add_partition_filters_and_store(
+        config_managers, partition_filters, config_dir, args
+    )
+
+
+def _get_primary_key_partition_filters(args: Namespace) -> List[List[str]]:
```

Comment:
We want to build this query using Ibis, not pandas ideally. This logic should also go in a new file query_builder/partitioned_row_builder.py or something similar. It will be very similar to random_row_builder where it [builds the query](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/query_builder/random_row_builder.py#L110) using Ibis.

For example, in Ibis you can do table.count() or table[primary_key].min() and then execute the query on source or target.